src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/external/W3SchoolClient.java

```diff
@@ -0,0 +1,43 @@
+package com.google.step.YOUR_PROJECT_NAME_HERE.external;
+
+import com.google.step.YOUR_PROJECT_NAME_HERE.data.Card;
+import java.io.IOException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.select.Elements;
+
+public final class W3SchoolClient {
+  private static final String TITLE_ID = "h1";
+  private static final String DESC_ID = "p";
+  private static final String SNIPPET_ID = "w3-example";
+  private static final String CODE_ID = "w3-code";
+
+  public Card search(String w3Link) throws IOException {
+
+    Document doc = Jsoup.connect(w3Link).get();
+    Elements titles = doc.getElementsByTag(TITLE_ID);
+    Elements descriptions = doc.getElementsByTag(DESC_ID);
+    Elements snippets = doc.getElementsByClass(SNIPPET_ID);
+
+    String title = "";
+    String description = "";
+    String code = "";
+
+    if (titles != null
+        && !titles.isEmpty()
+        && descriptions != null
```

Comment:
i would suggest doing null/empty check for each field separately.
```
if (titles == null || titles.isEmpty()) {
  return;
```

Reply:
*resolved via discussion*